modules/actions/analytics.js

```diff
@@ -99,13 +118,91 @@ export function getAppGeoDistribution(name, filterId) {
 							appName,
 						},
 					),
-				))
-			.catch(error => dispatch(
+				),
+			)
+			.catch((error) =>
+				dispatch(
 					createAction(
 						AppConstants.APP.ANALYTICS.GET_GEO_DISTRIBUTION_ERROR,
 						null,
 						error,
 					),
-				));
+				),
+			);
+	};
+}
+
+export function getAppAnalyticsInsights(name) {
+	return (dispatch, getState) => {
+		const appName = name || get(getState(), '$getCurrentApp.name', 'default');
+		dispatch(createAction(AppConstants.APP.ANALYTICS.GET_INSIGHTS));
+		return getAnalyticsInsights(appName)
+			.then((res) => {
+				if (res.status === 404) {
+					throw new Error('404: Insights route Not Found');
+				}
+				if (res.status >= 400) {
+					throw new Error(res.error);
+				} else {
+					dispatch(
+						createAction(
+							AppConstants.APP.ANALYTICS.GET_INSIGHTS_SUCCESS,
+							res,
+							undefined,
+							{
+								appName,
+							},
+						),
+					);
+				}
+			})
+			.catch((error) =>
+				dispatch(createAction(AppConstants.APP.ANALYTICS.GET_INSIGHTS_ERROR, null, error)),
+			);
+	};
+}
+
+/*
+	currentStatus: This is the current status of Insight - used in reducers for updating the data of the status.
+	nextStatus: This is the status in which we want to update the insight.
+	id: Insight Id
+*/
```

Comment:
@jyash97 Use JS format for docs to write the comments for params.

Reply:
Okay

>